tools/Flboundadapt.F90

```diff
@@ -0,0 +1,187 @@
+     input_geometryname_c, &
+     output_meshname_c)  bind(c)
+  !!< Peforms a mesh adapt based on the supplied input options file.
+  !!< Outputs the resulting mesh.
+ 
+  use iso_c_binding
+  use futils
+  use global_parameters, only: FIELD_NAME_LEN, OPTION_PATH_LEN
+  use fldebug
+  use parallel_tools, only: isparallel
+  use reference_counting
+  use spud
+  use fields
+  use edge_length_module
+  use state_module
+  use field_options
+  use vtk_interfaces
+  use diagnostic_fields_wrapper, only: calculate_diagnostic_variables
+  use limit_metric_module
+  use mesh_files
+  use Read_gmsh
+  use populate_state_module
+  use metric_assemble
+  use adapt_state_module
+  use diagnostic_fields_new, only : &
+    & calculate_diagnostic_variables_new => calculate_diagnostic_variables
+
+  implicit none
+  
+  interface
+    subroutine check_options()
+    end subroutine check_options
+
+#ifdef HAVE_PYTHON
+    subroutine python_init()
+    end subroutine python_init
+#endif
+  end interface
+  
+  type(c_ptr), value :: input_flmlname_c, input_geometryname_c,&
+       output_meshname_c
+
+  character(len=FIELD_NAME_LEN):: input_flmlname
+  character(len=FIELD_NAME_LEN):: input_geometryname
+  character(len=FIELD_NAME_LEN):: output_meshname
+  integer :: i, stat
+  type(mesh_type), pointer :: old_mesh
+  type(state_type), dimension(:), pointer :: states
+  type(vector_field) :: new_mesh_field
+  type(vector_field), pointer :: new_mesh_field_ptr, old_mesh_field
+  type(tensor_field) :: metric, t_edge_lengths
+  character(len=FIELD_NAME_LEN) :: mesh_format
+  character(len=OPTION_PATH_LEN) :: mesh_file_name
+  real, dimension(:,:,:), allocatable :: metric_data 
+
+  ! now turn into proper fortran string
+  call copy_c_string_to_fortran(input_flmlname_c, input_flmlname)
+  call copy_c_string_to_fortran(input_geometryname_c, input_geometryname)
+  call copy_c_string_to_fortran(output_meshname_c, output_meshname)
+  
+  ewrite(1, *) "In flboundadapt"
+ 
+#ifdef HAVE_PYTHON
+  call python_init()
+#endif
+
+  ewrite(2, *) "Input flml name: " // trim(input_flmlname)
+  ewrite(2, *) "Input geometry name: " // trim(input_geometryname)
+  ewrite(2, *) "Output mesh name: " // trim(output_meshname)
+
+  ! Load the options tree
+  call load_options(trim(input_flmlname) // ".flml")
+  if(.not. have_option("/simulation_name")) then
+    FLExit("Failed to find simulation name after loading options file")
+  end if
+  if(debug_level() >= 1) then
+    ewrite(1, *) "Options tree:"
+    call print_options()
+  end if
+#ifdef DDEBUG
+  ewrite(1, *) "Performing options sanity check"
+  call check_options()
+  ewrite(1, *) "Options sanity check successful"
+#endif
+
+  ! Populate the system state
+  call populate_state(states)
+
+  ! Calculate diagnostic fields
+  call calculate_diagnostic_variables(states)
+  call calculate_diagnostic_variables_new(states)
+
+  ! Find the external mesh field
+  !call find_mesh_field_to_adapt(states(1), old_mesh_field)
```

Comment:
Commented out code.
Should this perhaps be `topology_mesh_name` ? The metric doesn't necessarily live on the CoordinateMesh: periodic, extruded, higher order, etc. -  I realize not all of these combination may actually work with this tool of course. If Coordinate is the safest choice here, maybe add some comment why you think that is...